OpenRA.Mods.Common/Util.cs

```diff
@@ -175,5 +175,23 @@ public static int RandomDelay(World world, int[] range)
 
 			return world.SharedRandom.Next(range[0], range[1]);
 		}
+
+		static WDist allMinimumRequiredScanRadius;
+		static WDist blockerMinimumRequiredScanRadius;
+		static Ruleset lastRules = null;
+
+		public static WDist MinimumRequiredScanRadius(Ruleset rules, ActorInfo ai, bool onlyBlockers = false)
+		{
+			if (lastRules != rules)
```

Comment:
Ok, having multiple Health traits does complicate this.  My next question then is: do we need to cache this?  These calculations are done behind the loadscreen before the game starts, and probably takes no more than a few tens of ms all up without caching.